docs/downloader.rst

```diff
@@ -0,0 +1,157 @@
+
+.. -*- coding: utf-8 -*-
+
+.. _config:
+
+Downloader Design
+=================
+
+This describes code to accomplish remote to local synchronization in
+"the Leif Design" (see :ref:`Leif's Proposal: Magic-Folder
+"single-file" snapshot design`).
+
+We do not describe the synchronization of the list of participants in
+a magic-folder. This list will consist of at least:
+
+- an arbitrary ``name`` for each participant
+- a directory read-capability for each participant ("personal DMD")
+  - in this directory is a flattened representation of every file in
+    the Magic Folder (similar to the Tahoe 1.14.0 design) with each
+    entry pointing to a Snapshot. That is, there are no
+    sub-directories to recurse into, all files are listed by
+    downloading just the "personal DMD" capability of that participant.
+
+"A Snapshot" is a single version of a single file. It is represented
+by an immutable directory and contains:
```

Comment:
This data description is good.  I feel like it might belong somewhere else.  It's shared with the uploader and any other processes that want to interact with the data since they must all agree on it.  Perhaps a separate "model" or "data structures" document that holds this stuff would be good, then uploader, downloader, and anything else could reference it.